noir-projects/noir-protocol-circuits/crates/types/src/utils/arrays/assert_combined_array.nr

```diff
@@ -91,8 +91,10 @@ mod tests {
         }
 
         pub fn check_and_execute(self) {
-            let combined =
-                unsafe { combine_arrays(self.original_array_lt, self.original_array_gte) };
+            let combined = unsafe {
+                //@safety The value is checked against the combined array below.
+                combine_arrays(self.original_array_lt, self.original_array_gte)
+            };
```

Comment:
More tests, make unconstrained. 

Also this kind of pattern would never make sense, we'll never have

```noir
let foo = unsafe { compute_foo(bar); }
assert_eq(foo, some_value);
```

because then we'd just use `some_value`, there's no point in calling `compute_foo`. This is a dead giveaway that what we're doing is checking that `compute_foo` is returning an expected value, not constraining it. So the comment ends up being both wrong and misleading.